.github/workflows/maven.yml

```diff
@@ -21,5 +25,22 @@ jobs:
         with:
           distribution: 'temurin'
           java-version: '17'
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+
+      - name: Build and push image
```

Comment:
You should build the backend before building the docker (because the dockerfile make reference to something that do exist only after build)